lib/build.js

```diff
@@ -557,6 +557,28 @@ class BuildModel extends BaseModel {
         (this.status === 'UNSTABLE' && !!this.endTime);
     }
 
+    /**
+     * Remove all steps associated with this build and the build itself
+     * @return {Promise}        Resolves to null if remove successfully
+     */
+    remove() {
+        const removeSteps = (() =>
+            // Iterate through the steps and remove them
+            this.getSteps().then((steps) => {
+                if (steps.length === 0) {
+                    // Done removing steps
+                    return null;
+                }
+
+                return Promise.all(steps.map(step => step.remove()))
+                    .then(() => removeSteps());
```

Comment:
why is this being done recursively ? is `steps` dynamic ?

Reply:
previously we have default pagination. That's why it's recursive. Closed this anyway in favor of https://github.com/screwdriver-cd/screwdriver/issues/1614